### PR TITLE
Introduce cross-compiled releases 

### PR DESCRIPTION
**Pull request by tomkarw**
This is a build + release pipeline based off of the amazing ripgrep project's [workflow](https://github.com/BurntSushi/ripgrep/blob/master/.github/workflows/release.yml).

Closes https://github.com/helsing-ai/buffrs/issues/175.

### CODE REVIEW

**Comment by tomkarw** [comment]
@mara-schulke Did you have some specific compilation targets in mind?

**Comment by tomkarw** [comment]
I think it's done. I just don't know how to test without actually doing a release, we might have to do a test one and then quickly delete it afterwards.

**Comment by mara-schulke** [comment]
@tomkarw lets chat tomorrow!

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+      CROSS_VERSION: v0.2.5
+      # Emit backtraces on panics.
+      RUST_BACKTRACE: 1
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+        - build: linux
+          os: ubuntu-latest
+          rust: nightly
```
Comment (mara-schulke) [inline .github/workflows/release.yaml:68]: I would only compile with stable!

File: registry/tests/proptest-regressions/main.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Seeds for failure cases proptest has generated in the past. It is
```
Comment (mara-schulke) [inline registry/tests/proptest-regressions/main.txt:1]: What is this file?

File: ci/ubuntu-install-packages [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
```
Comment (mara-schulke) [inline ci/ubuntu-install-packages:1]: Can we move this into `.github`?

File: .github/workflows/registry.yml [hunk 4]
```diff
@@ -66,10 +66,11 @@ jobs:
...
       - run: cd registry && cargo llvm-cov --workspace --fail-under-lines "$MINIMUM_LINE_COVERAGE_PERCENT"
 
+# TODO: This already runs across registiry folder in cli.yml, maybe it's time to extract common.yml?
   typos:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
       - uses: Swatinem/rust-cache@v2
       - run: cargo install typos-cli || true
-      - run: cd registry && typos
```
Comment (mara-schulke) [inline .github/workflows/registry.yml:75]: Can we revert?

**Comment by tomkarw** [comment]
@mara-schulke Link to the results: https://github.com/helsing-ai/buffrs/releases/tag/untagged-f90bff9935014a85fc00

**Review by mara-schulke — APPROVED** [review]

**Pull request merged**